compiler/rustc_interface/src/passes.rs

```diff
@@ -663,13 +662,11 @@ fn write_out_deps(
     }
 }
 
-pub fn prepare_outputs(
-    sess: &Session,
-    krate: &ast::Crate,
-    boxed_resolver: &RefCell<BoxedResolver>,
-    crate_name: Symbol,
-) -> Result<OutputFilenames> {
+fn output_filenames(tcx: TyCtxt<'_>, (): ()) -> Result<&Arc<OutputFilenames>> {
+    let sess = tcx.sess;
     let _timer = sess.timer("prepare_outputs");
+    let krate = &*tcx.resolver_for_lowering(()).borrow().1;
```

Comment:
```suggestion
    let (_, krate) = &*tcx.resolver_for_lowering(()).borrow();
```
Destructuring almost always looks better than tuple fields.

Reply:
do the similar field usage in 7cc499bb2182c0e4f0468d80a1a5ce95d5fa331e count?